lib/Backend/Inline.cpp

```diff
@@ -2548,7 +2668,7 @@ bool Inline::InlineApplyTarget(IR::Instr *callInstr, const FunctionJITTimeInfo*
     const auto inlineCacheIndex = applyTargetLdOpnd->AsPropertySymOpnd()->m_inlineCacheIndex;
     const auto inlineeData = inlinerData->GetLdFldInlinee(inlineCacheIndex);
 
-    if (!isArrayOpndArgumentsObject || SkipCallApplyTargetInlining_Shared(callInstr, inlinerData, inlineeData, /*isApplyTarget*/ true, /*isCallTarget*/ false))
+    if ((!isArrayOpndArgumentsObject && (argsCount == 2)) || SkipCallApplyScriptTargetInlining_Shared(callInstr, inlinerData, inlineeData, /*isApplyTarget*/ true, /*isCallTarget*/ false))
```

Comment:
The caller does the argsCount == 1 checked after the call to this API. I'm calling this API to at least get the inlineeData for the target

---
In reply to: [85824508](https://github.com/Microsoft/ChakraCore/pull/1838#discussion_r85824508) [](ancestors = 85824508)